### ISSUE TITLE
Heap out of memory

### ISSUE THREAD

**Issue by rhettl** [alex94puchades/jsonapify#15]
Hello, 

I have a collection with a rather large array of data, `responses` in the below sample.

```javascript
const courseResource = new Resource(Course, {
  type:       'courses',
  id:         {
    value:    new Property('evalId'),
    writable: false
  },
  links:      {
    self: {
      value:    new Template('/courses/${evalId}'),
      writable: false,
    }
  },
  attributes: {
    evalId:      new Property('evalId'),
    questionSet: new Property('questionSet'),
    answerSet:   new Property('answerSet'),
    course:      new Property('course'),
    class:       new Property('class'),
    term:        new Property('term'),
    teacher:     new Property('teacher'),
    responses:   new Property('responses'),
  }
});
```

Due to the size of `responses` and the quantity of `courses` (only 1800+) the heap is filled when pulling the data. If I remove `responses` from the resource, all is well. In other programs I have to limit the output, but that means I don't know pagination data. 

I can think of two ways to solve the issue: 
1. not show `responses` in the `enumerate` path
2. limit responses and do pagination separately.

can you think of any other options? I'm not sure which will be best or how to do them?

Thank you.

**Comment by alex94cp** [comment]
Hi @rhettl, I'm really sorry but I haven't had the time to keep an eye on this project and somehow I missed this issue. 

Regarding 1, I remember seeing a proposal about default field inclusion (so, for example, you could set the `responses` field to not show within the enumerate middleware -not implemented yet in jsonapify- and select it with `?select=+responses` in your GET requests), but I think it was discarded (see [here](https://github.com/json-api/json-api/issues/20)).

Another solution would be to make `Response` itself a resource, and link it with the `Course` resource via a HasMany relation, but that may or may not make sense within your design.

I hope it helps.

**Comment by rhettl** [comment]
Thank you very much for the response. 

I moved on to another project due to difficulties with this issue and others. I had this same set of issues (more or less) with many of the other JSON API resources as well and decided I needed more experience with JSON API before I could properly decide if my problem truly lied in my design decisions or all the softwares I tried. 

When I circle back to this application, I will re-evaluate, but it is likely that JSON API's intended structure just doesn't like the design of my database and I need to redesign (again 😢 ). 

Thank you very much for the response back. Have a wonderful day!
